sharding/xorValidatorShuffler_test.go

```diff
@@ -980,3 +1039,52 @@ func createShufflerArgs(eligiblePerShard int, waitingPerShard int, nbShards uint
 	}
 	return args
 }
+
+func areMapsEqual(firstMap map[uint32][]Validator, secondMap map[uint32][]Validator) bool {
+	if len(firstMap) != len(secondMap) {
+		return false
+	}
+
+	for shardId := range firstMap {
+		if !areListsEqual(firstMap[shardId], secondMap[shardId]) {
+			return false
+		}
+	}
+
+	return true
+}
+
+func areListsEqual(firstList []Validator, secondList []Validator) bool {
+	if len(firstList) != len(secondList) {
+		return false
+	}
+
+	for i := 0; i < len(firstList); i++ {
+		if !bytes.Equal(firstList[i].PubKey(), secondList[i].PubKey()) {
+			return false
+		}
+	}
+
+	return true
+}
+
+func copyArgs(args ArgsUpdateNodes) ArgsUpdateNodes {
+	newArg := ArgsUpdateNodes{
+		Eligible: copyMap(args.Eligible),
+		Waiting:  copyMap(args.Waiting),
+		NewNodes: append(make([]Validator, 0), args.NewNodes...),
```

Comment:
If this function is really necessary, why not using copy function?

Reply:
seems easier to copy in one line instead of multiple lines

Reply:
`copy(newArg.NewNodes, args.NewNodes)` and removed that line from the ArgsUpdateNodes initialization

Reply:
also a initialization with the len has to be done first. fixed